src/script/jvm_instance.cpp

```diff
@@ -151,10 +161,52 @@ String JvmInstance::to_string(bool* r_valid) {
     return ScriptInstance::to_string(r_valid);
 }
 
-void JvmInstance::refcount_incremented() {}
+void JvmInstance::refcount_incremented() {
+    // This function should only be called when we know the object is a RefCounted. We directly reinterpret the pointer to it
+    RefCounted* ref = reinterpret_cast<RefCounted*>(owner);
+    int refcount = ref->get_reference_count();
+
+    if (refcount == 2) {
```

Comment:
shouldn't this be a >=? then?

Reply:
I switched to 2 because I want the function to only be called when we increment from 1 to 2. Bumping it to higher numbers is irrelevant. The pingpong only happens in the 1 <=> 2 transitions.